### PR TITLE
docs: 3944: local minio install

### PR DESCRIPTION
**Pull request by Gornoka**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend

### Describe the reason for change
https://github.com/heartexlabs/label-studio/issues/3944

#### What alternative approaches were there?

* using compose profiles instead of multi-file deployments

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?

- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

**Comment by Gornoka** [comment]
Forgot that I had some other changes made to the base compose file on my branch, if you dislike the logging and network-related changes I will remove them.

File: docker-compose.yml [hunk 1]
```diff
@@ -40,27 +54,38 @@ services:
...
+    #    command: >
+    #      -c ssl=on
+    #      -c ssl_cert_file=/var/lib/postgresql/certs/server.crt
+    #      -c ssl_key_file=/var/lib/postgresql/certs/server.key
     environment:
       - POSTGRES_HOST_AUTH_METHOD=trust
     volumes:
       - ${POSTGRES_DATA_DIR:-./postgres-data}:/var/lib/postgresql/data
       - ./deploy/pgsql/certs:/var/lib/postgresql/certs:ro
+    networks:
```
Comment (farioas) [inline docker-compose.yml]: I guess it will break existing deployments

File: docker-compose.yml [hunk 2]
```diff
@@ -1,32 +1,46 @@
 version: "3.9"
 
+# rotating logs, so they do not overflow
```
Comment (farioas) [inline docker-compose.yml]: Please remove this section

File: docker-compose.yml [hunk 3]
```diff
@@ -40,27 +54,38 @@ services:
       - POSTGRE_HOST=db
       - LABEL_STUDIO_HOST=${LABEL_STUDIO_HOST:-}
       - JSON_LOG=1
-#      - LOG_LEVEL=DEBUG
+    #      - LOG_LEVEL=DEBUG
     volumes:
       - ./mydata:/label-studio/data:rw
+    networks:
```
Comment (farioas) [inline docker-compose.yml]: Let's move network to override file

**Review by farioas — CHANGES_REQUESTED** [review]
There should be no changes in docker-compose file at all. Please move all to override

File: docker-compose.yml (hunk 2 above)
Comment (Gornoka) [reply docker-compose.yml]: can I keep them for the new components in docker-compose.minio.yaml?

File: docker-compose.yml (hunk 1 above)
Comment (Gornoka) [reply docker-compose.yml:82]: > There should be no changes in docker-compose file at all. Please move all to override

is it intended that no container restarts, except of postgres, which restarts on reboot, even if the stack was shutdown before ?

**Comment by Gornoka** [comment]
> There should be no changes in docker-compose file at all. Please move all to override

is it intended that no container restarts, except for Postgres, which restarts on reboot, even if the stack was stopped before?

File: docker-compose.yml (hunk 3 above)
Comment (Gornoka) [reply docker-compose.yml]: Having different networking approaches in the override and base files would make the networking only worse and harder to understand for users.
I would instead fall back to using the same networking structure you use in the base compose file -> everything in the default network as a free for all.

File: docker-compose.yml (hunk 2 above)
Comment (farioas) [reply docker-compose.yml]: yes

File: docker-compose.yml (hunk 3 above)
Comment (farioas) [reply docker-compose.yml]: ok, let's stick with existing approach

File: docker-compose.yml [hunk 4]
```diff
@@ -40,27 +41,27 @@ services:
...
+    #      -c ssl_cert_file=/var/lib/postgresql/certs/server.crt
+    #      -c ssl_key_file=/var/lib/postgresql/certs/server.key
     environment:
       - POSTGRES_HOST_AUTH_METHOD=trust
     volumes:
       - ${POSTGRES_DATA_DIR:-./postgres-data}:/var/lib/postgresql/data
       - ./deploy/pgsql/certs:/var/lib/postgresql/certs:ro
 
 volumes:
-  static: {}
```
Comment (farioas) [inline docker-compose.yml:66]: Actually this volume was deprecated, you can remove volumes section here and `static` from your override file

File: docs/source/guide/storedata.md [hunk 5]
```diff
@@ -49,6 +49,48 @@ When you start Label Studio using Docker Compose, you start it using a PostgreSQ
...
 ```
 
+## Minio Blob Storage
+Minio is a blob storage that is compatible with Amazon S3. You can use Minio to store your labeling tasks.
+
+### starting the containers
+For local development you can host a local minio server to emulate a S3 based production environment more closely.
+an example docker-compose file for this is available in the [label-studio repository](/docker-compose.minio.yml).
+
+To run minion with your lable studio instance you can use the following command:
```
Comment (farioas) [inline docs/source/guide/storedata.md]: run Minio with your Label Studio

File: docs/source/guide/storedata.md [hunk 6]
```diff
@@ -49,6 +49,48 @@ When you start Label Studio using Docker Compose, you start it using a PostgreSQ
...
+````
+The minio server will be available at http://localhost:9000.
+to configure minio settings you can create a .env file. Please remember to override the default credentials.
+
+````.dotenv
+MINIO_ROOT_USER=minio_admin_do_not_use_in_production
+MINIO_ROOT_PASSWORD=minio_admin_do_not_use_in_production
+````
+
+### connecting LS to local minio
```
Comment (farioas) [inline docs/source/guide/storedata.md]: Connect

File: docs/source/guide/storedata.md [hunk 7]
```diff
@@ -49,6 +49,48 @@ When you start Label Studio using Docker Compose, you start it using a PostgreSQ
...
+The following entry redirects all requests to minio to your local system:
+``127.0.0.1 minio``
+
+On Widows you can find your host file at `C:\Windows\System32\drivers\etc\hosts`
+On Linux you can find your host file at `/etc/hosts`
+On mac you can find your host file at `/private/etc/hosts`
+
+After altering your host file you can connect to your minio server with your browser at http://minio:9000.
+
+### removing minio data,
```
Comment (farioas) [inline docs/source/guide/storedata.md]: Remove Minio data

File: prometheus/prometheus.yml
Comment (farioas) [inline prometheus/prometheus.yml:1]: Let's move it to `prometheus/minio/prometheus.yml` so we'll be able to add more examples with prometheus in the future.

**Review by farioas — CHANGES_REQUESTED** [review]

**Comment by Gornoka** [comment]
@farioas is there something else needed to progress this PR?

**Comment by farioas** [comment]
Hi @Gornoka,

Nothing from your side. I have to test this change, rewrite the documentation and merge it. 
Unfortunatly, I'm too busy during this month, so I'll get back to this PR whenever it's possible.
Thanks.

**Comment by farioas** [comment]
Thanks @Gornoka !

**Pull request merged**